packages/lit-element/src/hydrate-support.ts

```diff
@@ -30,7 +31,49 @@ interface PatchableLitElement extends HTMLElement {
 }: {
   LitElement: PatchableLitElement;
 }) => {
-  // Capture whether we need hydration or not
+  const observedAttributes = Object.getOwnPropertyDescriptor(
+    Object.getPrototypeOf(LitElement),
+    'observedAttributes'
+  )!.get!;
+
+  // Add `defer-hydration` to observedAttributes
+  Object.defineProperty(LitElement, 'observedAttributes', {
+    get() {
+      return [...observedAttributes.call(this), 'defer-hydration'];
+    },
+  });
+
+  // Enable element when 'defer-hydration' attribute is removed by calling the
+  // super.connectedCallback()
+  const attributeChangedCallback =
+    LitElement.prototype.attributeChangedCallback;
+  LitElement.prototype.attributeChangedCallback = function (
+    name: string,
+    old: string | null,
+    value: string | null
+  ) {
+    if (name === 'defer-hydration' && value === null) {
+      connectedCallback.call(this);
+    } else {
+      attributeChangedCallback.call(this, name, old, value);
```

Comment:
I think you should always call into the original `attributeChangedCallback`. What if an element wanted to do something special on hydration?

Reply:
Yeah good call. Updated.